### PR TITLE
Build refactor

### PR DESCRIPTION
**Pull request by jsalazar**
Hi John -

This will require some review and discussion to describe my strategy.

Originally, this issue was to add in a javascript task, but when I got started, I realized it was better to tackled the whole file.

My goal became to provide the same functionality as before and clean up the build file.

- output fractal build to dist/  (let fractal do this natively)
- compile scss on save and update public/stylesheets/site.css
- start fractal
- delete dist (current CLEAN occurs on demand - change it to )

Build Scripts

- npm run styles --OR-- gulp styles
- npm run start   --OR-- gulp  (starts fractal)
- npm run build   --OR-- fractal build

Questions
- Do we currently need a Javascript task? We can add one once we nail down some requirements.
- do we need clean() task? 
We really don't need clean() due to ephemeral philosophy of build systems.
Build assets are meant to be temporary, so 'overwrite' is default
- Fractal deletes dist/ on start()
- Gulp overwrites /public/stylesheets/site.css
- We can add dist/ etc.. to gitignore if we don't want to commit them
- If piping or cache becomes an issue, then clean() becomes useful.

Closes: #23

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,125 +1,82 @@
...
+
+	const server = fractal.web.server({ sync: true });
+		  server.on('error', err => logger.error(err.message));
+
+	await server.start();
+	logger.success(`Fractal server is now running at ${server.url}`);
+}
+
+exports.styles = series(styles); // `npm run styles` OR `gulp styles`
+exports.default = series(clean, styles, startFractal); // `npm run start` OR `gulp`
```
Comment (garza) [inline gulpfile.js:82]: reading this, it looks like images and fonts is no longer called, will this break the one test image we have in the repo right now (content card example)?

File: gulpfile.js [hunk 2]
```diff
@@ -1,125 +1,82 @@
...
+    so we can reference it within (components/_preview.hbs) for a real-time preview.
+2. start fractal server and let it do its thing (watch, livereload, build)
+*/
+
+// copy generated.scss with dynamic imports for
+// all component sass files into public/css   so
+// we can use it in /components/_preview.hbs
+function styles() {
+	return src('src/scss/**/*.scss')
+    .pipe(plumber())
```
Comment (garza) [inline gulpfile.js:38]: let's go ahead and remove plumber()? We may need it further down the line for better error reporting, but it doesn't look like we're using it right now

File: package.json [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
+  "private": true,
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/utsa-asc/college-dls.git"
   },
   "scripts": {
-    "start": "gulp dev",
-    "build": "gulp",
+    "start": "gulp",
+    "styles": "gulp styles",
```
Comment (garza) [inline package.json:13]: makes sense, keeping the build as simple as possible (without adding it into gulp)++, and making the dev serve the default tasks makes more sense, I like!

File: gulpfile.js (hunk 2 above)
Comment (jsalazar) [reply gulpfile.js:38]: Agreed and good catch! I'll remove it ....

File: gulpfile.js (hunk 1 above)
Comment (jsalazar) [reply gulpfile.js:82]: Agghh - lol - that was sloppy of me.

The Gulp tasks for images() and font() was for meant for a "watch" that I failed to include in the default task.

Fractal handles all build  and generation of images and fonts, but I'm not sure if optimizations are applied to images and font files.

Now that I think of it, maybe we should keep these tasks in place to apply optimizations before Fractal grabs them for the build.

I'll add them to the default task.

File: package.json (hunk 3 above)
Comment (jsalazar) [reply package.json:13]: Yeah, the more I thought about it, the more I realized what role Gulp played in our overall build process.
I decided to keep it super simple and we can add tasks as needed as we get closer to the final build.

**Review by garza — APPROVED** [review]

**Review by garza-utsa — APPROVED** [review]
haha, was logged in as the wrong user =)

**Pull request merged**